src/main/java/Duke.java

```diff
@@ -1,10 +1,136 @@
+import java.util.Arrays;
+import java.util.Scanner;
+
+/**
+ * A personal assistant chatbot.
+ */
 public class Duke {
+    private static final String INDENTED_HORIZONTAL_LINE = " ".repeat(4) + "_".repeat(60);
+    /** Array of Task objects */
+    private static final Task[] tasks = new Task[100];
+    /** Number of tasks */
+    private static int taskCount = 0;
+
+    private static String indent(String text) {
+        String[] lines = text.split(System.lineSeparator());
+        for (int i = 0; i < lines.length; i++) {
+            lines[i] = " ".repeat(5) + lines[i];
+        }
+        return String.join(System.lineSeparator(), lines);
+    }
+
+    /**
+     * Prints out the specified text formatted as a response block.
+     * Horizontal lines will be printed before and after the specified text, and the text will be indented.
+     *
+     * @param text Text to be printed out.
+     */
+    private static void printResponseBlock(String text) {
+        System.out.println(INDENTED_HORIZONTAL_LINE);
+        System.out.println(indent(text));
+        System.out.println(INDENTED_HORIZONTAL_LINE);
+        System.out.println();
+    }
+
+    private static void printGreeting() {
+        String greeting = "Hello! I'm Duke" + System.lineSeparator()
+                + "What can I do for you?";
+        printResponseBlock(greeting);
+    }
+
+    /** Prints out the list of tasks (numbered) together with their status icons */
+    private static void printTasks() {
+        String[] formattedTasks = new String[taskCount];
```

Comment:
I like the naming of this variable which clearly distinguishes itself from normal tasks.